### PR TITLE
Feature/get proxied code (5)

### PR DESCRIPTION
**Pull request by jcortejoso**
Resolve https://github.com/celo-org/celo-monorepo/issues/995

## Motivation

The intention of this change is to visualize the source code/abi of contracts that are proxied. 

### Enhancements

* Modification of the `getsourcecode` RPC api allowing to retrieve the source code of a proxied contract when is called using the proxy address
* Modification of the contract source code view allowing to show the information of a proxied contract and an indicator that contract is proxied

## Upgrading

A new table `proxy_contract` is added. It's necessary to run the migration script to create the new table.
The proxy address and implementation address need to be registered in that table via INSERT sentence.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain.ex [hunk 1]
```diff
@@ -1016,7 +1019,7 @@ defmodule Explorer.Chain do
     |> join_associations(necessity_by_association)
     |> with_decompiled_code_flag(hash, query_decompiled_code_flag)
     |> Repo.one()
-    |> case do
+    |> case do:debugger.start()
```
Comment (nambrot) [inline apps/explorer/lib/explorer/chain.ex]: Should this be in here?

File: apps/explorer/lib/explorer/chain.ex [hunk 2]
```diff
@@ -2661,6 +2664,24 @@ defmodule Explorer.Chain do
     end
   end
 
+  #@spec get_proxied_address(Hash.Address.t()) :: ProxyContract.t() | nil
+  def get_proxied_address(address_hash) do
+    Logger.info("--getProxiedAddress: #{address_hash}")
```
Comment (nambrot) [inline apps/explorer/lib/explorer/chain.ex]: Should this stay in?

File: apps/explorer/lib/explorer/chain/proxy_contract.ex [hunk 3]
```diff
@@ -0,0 +1,56 @@
+defmodule Explorer.Chain.ProxyContract do
+  @moduledoc """
+  The representation of a Proxied Smart Contract.
+  "A contract in the sense of Solidity is a collection of code (its functions)
+  and data (its state) that resides at a specific address on the Ethereum
+  blockchain."
+  http://solidity.readthedocs.io/en/v0.4.24/introduction-to-smart-contracts.html
+  """
+
+  require Logger
```
Comment (nambrot) [inline apps/explorer/lib/explorer/chain/proxy_contract.ex]: Seems unused in here now?

File: apps/explorer/lib/explorer/chain/proxy_contract.ex [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  * `implementation_address` - address of the implementation contract behind the proxy.
+  """
+
+  @type t :: %Explorer.Chain.ProxyContract{
+               proxy_address: Hash.Address.t(),
+               implementation_address: Hash.Address.t()
+             }
+
+  schema "proxy_contract" do
+#    belongs_to(
```
Comment (nambrot) [inline apps/explorer/lib/explorer/chain/proxy_contract.ex]: Remove?

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 FROM bitwalker/alpine-elixir-phoenix:1.9.0
 
-RUN apk --no-cache --update add alpine-sdk gmp-dev automake libtool inotify-tools autoconf python
+RUN apk --no-cache --update add alpine-sdk gmp-dev automake libtool inotify-tools autoconf python
```
Comment (nambrot) [inline docker/Dockerfile]: Whitespace

File: apps/explorer/lib/explorer/chain.ex [hunk 6]
```diff
@@ -2661,6 +2664,24 @@ defmodule Explorer.Chain do
...
+    query =
+      from(contract in ProxyContract,
+        where: contract.proxy_address == ^address_hash
+      )
+    Logger.info("--after query")
+
+    #Repo.one(query)
```
Comment (nambrot) [inline apps/explorer/lib/explorer/chain.ex]: Remove

File: apps/explorer/lib/explorer/chain.ex (hunk 1 above)
Comment (aaitor) [reply apps/explorer/lib/explorer/chain.ex]: fixed

File: apps/explorer/lib/explorer/chain.ex (hunk 2 above)
Comment (aaitor) [reply apps/explorer/lib/explorer/chain.ex]: fixed

File: apps/explorer/lib/explorer/chain/proxy_contract.ex (hunk 3 above)
Comment (aaitor) [reply apps/explorer/lib/explorer/chain/proxy_contract.ex]: removed

File: docker/Dockerfile (hunk 5 above)
Comment (aaitor) [reply docker/Dockerfile]: removed

File: apps/explorer/lib/explorer/chain.ex (hunk 6 above)
Comment (aaitor) [reply apps/explorer/lib/explorer/chain.ex]: removed

**Pull request merged**